Can't run `@QuarkusIntegrationTest` starting with Quarkus 3.31.3

### Describe the bug

Running `@QuarkusIntegrationTest`s starting with Quarkus 3.31.3 fails. Its fine 3.31.2 and before.

### Expected behavior

I would expect `@QuarkusIntegrationTest`s to run without issue.

### Actual behavior

`@QuarkusIntegrationTest` execution is failing with the following error:

```
Error: Could not find or load main class "-Dquarkus-internal-test.serialized-app-model.path=.Users.edeandre.workspaces.quarkus.quarkus-super-heroes.ui-super-heroes.target.quarkus.bootstrap.test-app-model.dat"
Caused by: java.lang.ClassNotFoundException: "-Dquarkus-internal-test.serialized-app-model.path=.Users.edeandre.workspaces.quarkus.quarkus-super-heroes.ui-super-heroes.target.quarkus.bootstrap.test-app-model.dat"
Process exited with non-zero status: 1
```

Here is the full log:

```
[INFO] --- failsafe:3.5.4:integration-test (default) @ rest-narration ---
[INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] ---io.quarkus.sample.superheroes.narration.rest.AzureOpenAiNarrationResourceIT - 0.001 s
[INFO]    +-- [??] helloEndpoint (Azure OpenAI profile is not enabled) - 0 s
[INFO]    +-- [??] invalidFightToNarrate (Azure OpenAI profile is not enabled) - 0 s
[INFO]    +-- [??] shouldPingOpenAPI (Azure OpenAI profile is not enabled) - 0 s
[INFO]    +-- [??] invalidNarrationToFetchImage (Azure OpenAI profile is not enabled) - 0 s
[INFO]    +-- [??] shouldGenerateAnImageFromNarration (Azure OpenAI profile is not enabled) - 0 s
[INFO]    +-- [??] shouldNarrateAFight (Azure OpenAI profile is not enabled) - 0 s
[INFO]    '-- [??] shouldGetAFallbackOnError (Azure OpenAI profile is not enabled) - 0 s
18:51:22 INFO  [io.qu.wi.de.WireMockServer] (build-2) Using HTTP server impl: Jetty11HttpServer
18:51:22 INFO  [io.qu.wi.de.WireMockServer] (qtp1135738744-70) Admin request received:
127.0.0.1 - GET /settings

Host: [localhost:55894]
Connection: [keep-alive]
User-Agent: [Apache-HttpClient/5.5.1 (Java/21.0.9)]


Executing "/Users/edeandre/.sdkman/candidates/java/21.0.9-tem/bin/java --add-opens=java.base/java.lang.invoke=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-exports=java.base/jdk.internal.module=ALL-UNNAMED "-Dquarkus-internal-test.serialized-app-model.path=/Users/edeandre/workspaces/quarkus/quarkus-super-heroes/rest-narration/target/quarkus/bootstrap/test-app-model.dat" -Dquarkus.http.port=0 -Dquarkus.http.ssl-port=8444 -Dtest.url=http://localhost:${quarkus.http.test-port:8081} -Dquarkus.log.file.path=/Users/edeandre/workspaces/quarkus/quarkus-super-heroes/rest-narration/target/quarkus.log -Dquarkus.log.file.enabled=true -Dquarkus.log.category."io.quarkus".level=INFO -Dquarkus.profile=prod -Dquarkus.langchain4j.openai.dalle3.log-responses=true -Dquarkus.langchain4j.openai.max-retries=2 -Dquarkus.wiremock.devservices.port=55894 -Dquarkus.platform.version=3.31.3 -Dquarkus.langchain4j.openai.log-responses=true -Dquarkus.langchain4j.openai.base-url=http://localhost:${quarkus.wiremock.devservices.port}/v1/ -Dquarkus.langchain4j.openai.dalle3.max-retries=2 -Dquarkus.langchain4j.openai.dalle3.timeout=3s -Dquarkus.config.build-time-mismatch-at-runtime=fail -Dquarkus.langchain4j.openai.dalle3.enable-integration=true -Dquarkus.langchain4j.openai.dalle3.log-requests=true -Dquarkus.langchain4j.openai.timeout=3s -Dquarkus.test.arg-line=--add-opens=java.base/java.lang.invoke=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-exports=java.base/jdk.internal.module=ALL-UNNAMED "-Dquarkus-internal-test.serialized-app-model.path=/Users/edeandre/workspaces/quarkus/quarkus-super-heroes/rest-narration/target/quarkus/bootstrap/test-app-model.dat" -Dquarkus.langchain4j.openai.log-requests=true -Dquarkus.langchain4j.openai.dalle3.base-url=http://localhost:${quarkus.wiremock.devservices.port}/v1/ -Dquarkus.langchain4j.openai.enable-integration=true -jar /Users/edeandre/workspaces/quarkus/quarkus-super-heroes/rest-narration/target/quarkus-app/quarkus-run.jar"
Error: Could not find or load main class "-Dquarkus-internal-test.serialized-app-model.path=.Users.edeandre.workspaces.quarkus.quarkus-super-heroes.rest-narration.target.quarkus.bootstrap.test-app-model.dat"
Caused by: java.lang.ClassNotFoundException: "-Dquarkus-internal-test.serialized-app-model.path=.Users.edeandre.workspaces.quarkus.quarkus-super-heroes.rest-narration.target.quarkus.bootstrap.test-app-model.dat"
Process exited with non-zero status: 1
[INFO] ---io.quarkus.sample.superheroes.narration.rest.OpenAiNarrationResourceIT - 1.956 s
[INFO]    +-- [XX] helloEndpoint - 0.004 s
[INFO] [*] helloEndpoint
[INFO] 
[INFO] Stack trace
[INFO] java.lang.RuntimeException: java.lang.RuntimeException: Unable to successfully launch process '13207'. Exit code is: '1'.
        at io.quarkus.test.junit.QuarkusIntegrationTestExtension.throwBootFailureException(QuarkusIntegrationTestExtension.java:380)
        at io.quarkus.test.junit.QuarkusIntegrationTestExtension.beforeEach(QuarkusIntegrationTestExtension.java:123)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: java.lang.RuntimeException: Unable to successfully launch process '13207'. Exit code is: '1'.
        at io.quarkus.test.common.LauncherUtil.ensureProcessIsAlive(LauncherUtil.java:130)
        at io.quarkus.test.common.LauncherUtil.waitForCapturedListeningData(LauncherUtil.java:93)
        at io.quarkus.test.common.DefaultJarLauncher.start(DefaultJarLauncher.java:82)
        at io.quarkus.test.junit.IntegrationTestUtil.startLauncher(IntegrationTestUtil.java:160)
        at io.quarkus.test.junit.QuarkusIntegrationTestExtension.doProcessStart(QuarkusIntegrationTestExtension.java:323)
        at io.quarkus.test.junit.QuarkusIntegrationTestExtension.ensureStarted(QuarkusIntegrationTestExtension.java:182)
        at io.quarkus.test.junit.QuarkusIntegrationTestExtension.beforeAll(QuarkusIntegrationTestExtension.java:142)
        ... 1 more

[INFO] 
[INFO] Standard out
18:51:22 INFO  [io.qu.wi.de.WireMockServer] (build-2) Using HTTP server impl: Jetty11HttpServer
18:51:22 INFO  [io.qu.wi.de.WireMockServer] (qtp1135738744-70) Admin request received:
127.0.0.1 - GET /settings

Host: [localhost:55894]
Connection: [keep-alive]
User-Agent: [Apache-HttpClient/5.5.1 (Java/21.0.9)]


Executing "/Users/edeandre/.sdkman/candidates/java/21.0.9-tem/bin/java --add-opens=java.base/java.lang.invoke=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-exports=java.base/jdk.internal.module=ALL-UNNAMED "-Dquarkus-internal-test.serialized-app-model.path=/Users/edeandre/workspaces/quarkus/quarkus-super-heroes/rest-narration/target/quarkus/bootstrap/test-app-model.dat" -Dquarkus.http.port=0 -Dquarkus.http.ssl-port=8444 -Dtest.url=http://localhost:${quarkus.http.test-port:8081} -Dquarkus.log.file.path=/Users/edeandre/workspaces/quarkus/quarkus-super-heroes/rest-narration/target/quarkus.log -Dquarkus.log.file.enabled=true -Dquarkus.log.category."io.quarkus".level=INFO -Dquarkus.profile=prod -Dquarkus.langchain4j.openai.dalle3.log-responses=true -Dquarkus.langchain4j.openai.max-retries=2 -Dquarkus.wiremock.devservices.port=55894 -Dquarkus.platform.version=3.31.3 -Dquarkus.langchain4j.openai.log-responses=true -Dquarkus.langchain4j.openai.base-url=http://localhost:${quarkus.wiremock.devservices.port}/v1/ -Dquarkus.langchain4j.openai.dalle3.max-retries=2 -Dquarkus.langchain4j.openai.dalle3.timeout=3s -Dquarkus.config.build-time-mismatch-at-runtime=fail -Dquarkus.langchain4j.openai.dalle3.enable-integration=true -Dquarkus.langchain4j.openai.dalle3.log-requests=true -Dquarkus.langchain4j.openai.timeout=3s -Dquarkus.test.arg-line=--add-opens=java.base/java.lang.invoke=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-exports=java.base/jdk.internal.module=ALL-UNNAMED "-Dquarkus-internal-test.serialized-app-model.path=/Users/edeandre/workspaces/quarkus/quarkus-super-heroes/rest-narration/target/quarkus/bootstrap/test-app-model.dat" -Dquarkus.langchain4j.openai.log-requests=true -Dquarkus.langchain4j.openai.dalle3.base-url=http://localhost:${quarkus.wiremock.devservices.port}/v1/ -Dquarkus.langchain4j.openai.enable-integration=true -jar /Users/edeandre/workspaces/quarkus/quarkus-super-heroes/rest-narration/target/quarkus-app/quarkus-run.jar"
[INFO] 
[INFO] Standard error
Error: Could not find or load main class "-Dquarkus-internal-test.serialized-app-model.path=.Users.edeandre.workspaces.quarkus.quarkus-super-heroes.rest-narration.target.quarkus.bootstrap.test-app-model.dat"
Caused by: java.lang.ClassNotFoundException: "-Dquarkus-internal-test.serialized-app-model.path=.Users.edeandre.workspaces.quarkus.quarkus-super-heroes.rest-narration.target.quarkus.bootstrap.test-app-model.dat"
Process exited with non-zero status: 1
[INFO]    +-- [??] invalidFightToNarrate - 0 s
[INFO]    +-- [??] shouldPingOpenAPI - 0 s
[INFO]    +-- [??] invalidNarrationToFetchImage - 0 s
[INFO]    +-- [??] shouldGenerateAnImageFromNarration - 0 s
[INFO]    +-- [??] shouldNarrateAFight - 0 s
[INFO]    '-- [??] shouldGetAFallbackOnError - 0 s
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   OpenAiNarrationResourceIT.helloEndpoint » Runtime java.lang.RuntimeException: Unable to successfully launch process '13207'. Exit code is: '1'.
[INFO] 
[ERROR] Tests run: 14, Failures: 0, Errors: 1, Skipped: 13
```

### How to Reproduce?

1. Clone https://github.com/quarkusio/quarkus-super-heroes
2. `cd rest-narration`
3. `./mvnw clean verify -Dquarkus.platform.version=3.31.3`

See the error

### Output of `uname -a` or `ver`

`Darwin edeandrea-m4max 25.3.0 Darwin Kernel Version 25.3.0: Wed Jan 28 20:48:41 PST 2026; root:xnu-12377.81.4~5/RELEASE_ARM64_T6041 arm64`

### Output of `java -version`

```
openjdk version "21.0.9" 2025-10-21 LTS
OpenJDK Runtime Environment Temurin-21.0.9+10 (build 21.0.9+10-LTS)
OpenJDK 64-Bit Server VM Temurin-21.0.9+10 (build 21.0.9+10-LTS, mixed mode, sharing)
```

### Quarkus version or git rev

3.31.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

```
Apache Maven 3.9.9 (8e8579a9e76f7d015ee5ec7bfcdc97d260186937)
Maven home: /Users/edeandre/.m2/wrapper/dists/apache-maven-3.9.9/3477a4f1
Java version: 21.0.9, vendor: Eclipse Adoptium, runtime: /Users/edeandre/.sdkman/candidates/java/21.0.9-tem
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "26.3", arch: "aarch64", family: "mac"
```

### Additional information

_No response_

--- a/devtools/maven/src/main/java/io/quarkus/maven/GenerateCodeMojo.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/GenerateCodeMojo.java
@@ -122,7 +122,7 @@ public class GenerateCodeMojo extends QuarkusBootstrapMojo {
                             Properties properties = mavenProject().getProperties();
                             String argLine = properties.getProperty("argLine", "");
                             properties.setProperty("argLine", argLine +
-                                    " \"-D" + BootstrapConstants.SERIALIZED_TEST_APP_MODEL + "=" + serializedTestAppModelPath
+                                    " -D" + BootstrapConstants.SERIALIZED_TEST_APP_MODEL + "=\"" + serializedTestAppModelPath
                                     + "\"");
                         } catch (IOException e) {
                             getLog().warn("Failed to serialize application model", e);

